### PR TITLE
steps: add version 4.1.1, remove others

### PR DESCRIPTION
**Pull request by tristan0x**
* only keep the maintained version
* refresh the `cmake_args` function
* add maintainer

### CODE REVIEW

**Comment by tristan0x** [comment]
How can a reviewer be assigned to this contribution?

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py]: What does this imply? Usually portability is semantically attached to the `target` attribute of a node, and never to a variant.

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py:16]: Usually we deprecate versions first, but this is fine since the package has no dependents. :+1:

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py:39]: I see you have a custom build type. Curious why are you defaulting to `RelWithDebInfo`.

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py]: Generator is now a directive, se #35552

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py]:
```suggestion
    variant("distmesh", default=True, when="+mpi", description="Add solvers based on distributed mesh")
```

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py]: This is implied if you use the `generator` directive
```suggestion
```

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py:81]: Given the comment, is this only needed when `~bundle`?

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py]:
```suggestion
        python_interpreter = self.spec["python"].command
```

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py]: Can you use `self.define` also here, for consistency?

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/steps/package.py:117]: FYI, this can be added as a directive, once #36286 is merged. The directive would be:
```python
requires("%gcc", when="+coverage", msg="GCC is required to build with coverage")
```

**Review by alalazo — CHANGES_REQUESTED** [review]
I left a few comments. Let me know if anything is unclear. Thanks!

File: var/spack/repos/builtin/packages/steps/package.py
Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py:39]: we keep the default value as defined https://github.com/spack/spack/blob/44196085f6613169f2c7fb8a772147eba0ef1650/lib/spack/spack/build_systems/cmake.py#L93-L98

The new `RelWithDebInfoAndAssert` CMake build type is used by our continuous integration process.

File: var/spack/repos/builtin/packages/steps/package.py
Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py]: Done in fcc66e8ef2
Thank you for the suggestion

File: var/spack/repos/builtin/packages/steps/package.py
Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py]: taken care of in 9fd95c1820
Thank you!

File: var/spack/repos/builtin/packages/steps/package.py
Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py:117]: Very nice improvement indeed. Should we wait for #36286 to be merged?

File: var/spack/repos/builtin/packages/steps/package.py
Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py:81]: This is needed in both cases. This piece of dirt will be cleaned up in the next release.

File: var/spack/repos/builtin/packages/steps/package.py
Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py]: In our HPC system, we have builds of this package that need to run on multiple x86 families (knl, ...). This variant is used for that, and passed as a CMake variable to the project. I don't know how it could be specified in a prettier way.

Comment (alalazo) [reply var/spack/repos/builtin/packages/steps/package.py]: What does the variant do, effectively? Does it just add `-march=native -mtune=native` or something more?

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [reply var/spack/repos/builtin/packages/steps/package.py:117]: Not really, but you can keep an eye on PRs, and submit an improvement to this recipe once that happens!

File: var/spack/repos/builtin/packages/steps/package.py
Comment (alalazo) [reply var/spack/repos/builtin/packages/steps/package.py]: :point_up: That's the only open question on my side. Other than that this PR LGTM.

Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py]: Yes, it is doing nothing more than that.

Comment (alalazo) [reply var/spack/repos/builtin/packages/steps/package.py]: Ok, then it should be already taken care of by our compiler wrapper. Depending on the `target` attribute, we inject the appropriate flags - so I think this variant can be removed. For a quick example of what we inject, you can try running:
```console
$ spack -d install zlib
```
At the end you should find two files similar to:
```
-rw-rw-r--  1 culpo culpo  11K apr 28 14:43 spack-cc-zlib-4kpu5pp.out.log
-rw-rw-r--  1 culpo culpo 5,2K apr 28 14:43 spack-cc-zlib-4kpu5pp.in.log
```
in your current working directory. The `*out.log` contains all the compilation flags expanded, and will look like:
```console
[cc] /usr/bin/gcc -march=icelake-client -mtune=icelake-client -c ztest21237.c
[cc] /usr/bin/gcc -march=icelake-client -mtune=icelake-client -c -fPIC -O2 ztest21237.c
...
```

Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py]: Thank you very much for your time and explanations. Indeed this `native` variant is not needed. I removed it in 7cd33e2718
Cheers

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Pull request merged**
